sql/sql_alter.cc

```diff
@@ -84,6 +84,24 @@ bool Alter_info::set_requested_lock(const LEX_STRING *str)
   return false;
 }
 
+/**
+  Checks if there are any columns with COLUMN_FORMAT COMRPESSED
+  attribute among field definitions in create_list.
+
+  @retval false there are no compressed columns
+  @retval true there is at least one compressed column
+*/
+bool Alter_info::has_compressed_columns() const
+{
+  List_iterator<Create_field> it(
+    const_cast<List<Create_field>&>(create_list));
+  Create_field* sql_field = it++;
```

Comment:
Can the list be empty so that iterator already points to one-past-list-end here, before we post-increment? I.e. why not idiomatic iteration loop?